### ISSUE TITLE
sen2cor() not working anymore

### ISSUE THREAD

**Issue by Weemaan** [ranghetti/sen2r#311]
Hello,

I was working with this package a couple of months ago and everything worked fine. A few days ago I was running my script again with the same data but some weird output from the sen2cor() function occured. 

These are my files from the l1c folder:
    S2A_MSIL1C_20170722T153911_N0205_R011_T18TYS_20170722T154309.SAFE
    S2A_MSIL1C_20170722T153911_N0205_R011_T18TYT_20170722T154309.SAFE
    S2A_MSIL1C_20170722T153911_N0205_R011_T19TCM_20170722T154309.SAFE
    S2A_MSIL1C_20170722T153911_N0205_R011_T19TCN_20170722T154309.SAFE

    sen2cor(l1c_prodlist = dir(file.path(workdir, region, "sen2r", "l1c")),
            l1c_dir =  file.path(workdir, region, "sen2r", "l1c"),
            outdir = file.path(workdir, region, "sen2r", "l2a"),
            proc_dir = file.path(workdir, region, "sen2r", "proc"),
            tmpdir = file.path(workdir, region, "sen2r", "tmpdir"),
            parallel = F)

Output:
[2020-05-03 14:34:34] Launching Sen2Cor on S2A_MS~1.SAF (1/4)... 

[2020-05-03 14:34:35] Running Sen2Cor on S2A_MS~1.SAF finished after 0.768 secs. 

[2020-05-03 14:35:22] Launching Sen2Cor on S2A_MS~2.SAF (2/4)... 

[2020-05-03 14:35:24] Running Sen2Cor on S2A_MS~2.SAF finished after 2.26 secs. 

[2020-05-03 14:35:59] Launching Sen2Cor on S2A_MS~3.SAF (3/4)... 

[2020-05-03 14:36:01] Running Sen2Cor on S2A_MS~3.SAF finished after 1.72 secs. 

[2020-05-03 14:36:52] Launching Sen2Cor on S2A_MS~4.SAF (4/4)... 

[2020-05-03 14:36:52] Running Sen2Cor on S2A_MS~4.SAF finished after 0.587 secs. 

So these "S2A_MS~1.SAF" files seem wrong. 

I have reinstalled python, sen2cor, R, RStudio and OSGeo4W allready.

edit:
R version 3.6.3 (2020-02-29)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

Package Version: 1.3.4

**Comment by ranghetti** [comment]
Hello,
I need another information in order to suggest you what you could try (#186, point 5).

**Comment by Weemaan** [comment]
Thanks for the quick reply!

        sen2r::load_binpaths()
        $aria2c
        [1] "C:\\Users\\SEBAST~1\\DOCUME~1\\Sen2R_Dep\\aria2c.exe"

        $gdalinfo
        [1] "C:\\OSGEO4~1\\bin\\gdalinfo.exe"

        $python
        [1] "C:\\OSGEO4~1\\bin\\python.exe"

        $gdal_calc
        [1] "C:\\OSGEO4~1\\bin\\python.exe C:\\OSGEO4~1\\bin\\GDAL_C~1.PY"

        $gdal_fillnodata
        [1] "C:\\OSGEO4~1\\bin\\python.exe C:\\OSGEO4~1\\bin\\GDAL_F~1.PY"

        $sen2cor
        [1] "C:\\Users\\SEBAST~1\\DOCUME~1\\SEN2R_~1\\Sen2Cor\\SEN2CO~1.05-\\L2A_PR~1.BAT"

        attr(,"path")
        [1] "C:\\Users\\SEBAST~1\\DOCUME~1\\SEN2R~1/paths.json"

Note that `SEN2R_~1` (..\Documents\Sen2R_Dep) is a different folder than `SEN2R~1` (..\Documents\.sen2r).

**Comment by ranghetti** [comment]
What error do you receive launching the followingo command in a cmd prompt?
```
C:\Users\SEBAST~1\DOCUME~1\SEN2R_~1\Sen2Cor\SEN2CO~1.05-\L2A_PR~1.BAT <your_l1c_path>\S2A_MSIL1C_20170722T153911_N0205_R011_T18TYS_20170722T154309.SAFE
```
(replacing `<your_l1c_path>` with your L1c path (better if in the short form).

Moreover, please verify and report the size of the L1C folders, as well as the content of the product `S2A_MSIL1C_20170722T153911_N0205_R011_T18TYS_20170722T154309.SAFE`.

**Comment by Weemaan** [comment]
Running the command with the short form created with the `sen2r::normalize_path()` function:

    C:\Users\SEBAST~1\DOCUME~1\sen2cor\SEN2CO~1.05-\L2A_PR~1.BAT C:\Users\SEBAST~1\DOCUME~1\R_SCRI~1\Projects\PURE_C~1\Regions\Quebec\sen2r\l1c\S2A_MS~1.SAF 
returns:
        `L1C user product directory must match the following mask: S2?_MSIL1C*.SAFE
but is: S2A_MS~1.SAF`

However if I use full path name of the .SAFE file:

    C:\Users\SEBAST~1\DOCUME~1\sen2cor\SEN2CO~1.05-\L2A_PR~1.BAT C:\Users\Sebastian\Documents\R_Scripts\Projects\PURE_Canada\Regions\Quebec\sen2r\l1c\S2A_MSIL1C_20170722T153911_N0205_R011_T18TYS_20170722T154309.SAFE

it seems to work just fine:
`Sentinel-2 Level 2A Processor (Sen2Cor). Version: 2.5.5, created: 2018.03.19, supporting Level-1C product version <= 14.5 started ...
Product version: 14.2. Operation mode: TOOLBOX. Processing baseline: 02.05.
 No resolution specified, will process all resolutions.` (and than the processing steps start working)

Also the output folder looks correct to me (`S2A_MSIL2A_20170722T153911_N0205_R011_T18TYS_20170722T154309.SAFE`)


My Sentinel 2 object (772 MB) has been downloaded from EarthExplorer and looks like this:

` [2] "AUX_DATA"                                              
 [3] "DATASTRIP"                                                 
 [4] "DATASTRIP/DS_MTI__20170722T205117_S20170722T154309"        
 [5] "DATASTRIP/DS_MTI__20170722T205117_S20170722T154309/QI_DATA"
 [6] "GRANULE"                                                   
 [7] "GRANULE/L1C_T18TYS_A010876_20170722T154309"                
 [8] "GRANULE/L1C_T18TYS_A010876_20170722T154309/AUX_DATA"       
 [9] "GRANULE/L1C_T18TYS_A010876_20170722T154309/IMG_DATA"       
[10] "GRANULE/L1C_T18TYS_A010876_20170722T154309/QI_DATA"        
[11] "HTML"                                                      
[12] "rep_info"`

**Comment by Weemaan** [comment]
Allright so I looked into your sen2cor() function and when I delete the normalize_path() part in lines 71&72 it works kinda. So I believe the probleme is there.

**Comment by ranghetti** [comment]
Thank you @Weemaan; I added the use of `shortPathName()` on SAFE paths to solve previous problems with long paths and spaces, but actually the SAFE name can not be shortened.
I added a patch on a development branch: can you try if it solves the problem?
```r
remotes::install_github("ranghetti/sen2r", ref = "hotfix/issue311")
```

**Comment by Weemaan** [comment]
Yes, this works perfectly now! Thanks a lot!
I'll keep searching what went wrong on my end that the SAFE name has been shortened.

Can I continue using this version in the future?

**Comment by ranghetti** [comment]
Thhanks to you @Weemaan for your input. The bug was fixed starting from GitHub version 1.3.4.9001. Users which need to use Sen2Cor on Windows are encouraged to install the GitHub version:
```r
remotes::install_github("ranghetti/sen2r", ref = "hotfix/issue311")
```
(I am leaving this issue open until version 1.3.5 is released on CRAN)

**Comment by ranghetti** [comment]
> Can I continue using this version in the future?

You can both use this version or update (in future) to a newer GitHub version (with the command shown in the previous post) or a newer CRAN version (starting from future version 1.3.5).
